### ISSUE TITLE
Query parameters are being URL-decoded when making a request

### ISSUE THREAD

**Issue by coyotte508** [fafhrd91/aiohttp#7222]
### Describe the bug

When making a request, eg to https://huggingface.co/?response-content-type=application%2Fgzip, the URL gets converted to https://huggingface.co/?response-content-type=application/gzip before making the request: the `%2F` is converted to `/` in the query.

Most of the time, this is harmless. But when making calls to cloudfront on a signed URL, the signature won't match anymore and cloudfront will answer with a 403.

### To Reproduce

```
import aiohttp
import asyncio

async def main():
    async with aiohttp.ClientSession() as session:
        async with session.get('https://huggingface.co/?response-content-type=application%2Fgzip') as response:
            print("Status:", response.status)
            print("Url:", response.url)

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

### Expected behavior

The code above should show

```
Status: 200
Url: https://huggingface.co/?response-content-type=application%2Fgzip
```

### Logs/tracebacks

```python-traceback
Status: 200
Url: https://huggingface.co/?response-content-type=application/gzip
```

Serverside, we also see that it's called with this decoded url

### Python Version

```console
Python 3.10.6
```

### aiohttp Version

```console
Name: aiohttp
Version: 3.8.4
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache 2
Location: /usr/local/lib/python3.10/dist-packages
Requires: aiosignal, async-timeout, attrs, charset-normalizer, frozenlist, multidict, yarl
Required-by:
```

### multidict Version

```console
$ python -m pip show multidict
```

### yarl Version

```console
Name: multidict
Version: 6.0.4
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /usr/local/lib/python3.10/dist-packages
Requires: 
Required-by: aiohttp, yarl
```

### OS

Linux

### Related component

Client

### Additional context

We use `aoihttp` in https://github.com/huggingface/datasets/ . Recently on hf.co, we added `Content-Type` to the redirected response when fetching LFS files, eg on https://huggingface.co/bert-base-uncased/resolve/main/flax_model.msgpack and it caused 403s to happen.

We removed the `response-content-type` we pass to cloudfront for now due to the problem, and there is a workaround (cloudfront allow to sign query params with a `*`) - so we're going to readd it and be fine. But it'd be nice to fix the issue here :)

Related issue: https://github.com/aio-libs/aiohttp/issues/4972 , it's serverside but probably related (yarl)

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by Dreamsorcerer** [comment]
I think you just want a pre-encoded URL:
https://github.com/aio-libs/aiohttp/issues/4307#issuecomment-548699254

**Comment by coyotte508** [comment]
Thanks for the related issue. In our case, the web server redirects to that url, so it implies doing manual redirects & complexifying the codebase.

**Comment by Dreamsorcerer** [comment]
Then I think it's a duplicate of #5319.
